src/main/java/com/google/testing/compile/JavaFileObjects.java

```diff
@@ -57,7 +58,8 @@ private JavaFileObjects() { }
    * <p>Note that this method makes no attempt to verify that the name matches the contents of the
    * source and compilation errors may result if they do not match.
    */
-  public static JavaFileObject forSourceString(String fullyQualifiedName, String source) {
+  public static JavaFileObject forSourceString(String fullyQualifiedName,
+      @Language("JAVA") String source) {
```

Comment:
Even in Java I find this overload to be superior than the varargs one. By preferring to use multiline concatenations, IntelliJ will let you edit it as a single chunk, you can copy/paste multi-line snippets in and out, and syntax highlighting will mostly work (which is what this enables automatically, but you previously could do manually).

Reply:
+1. Though in going through to make a gif example to share here for demoing, I noticed that adding this language annotation here actually throws off the multiline vararg option :/.

![image](https://user-images.githubusercontent.com/1361086/55219054-1555e700-51c1-11e9-8a0f-48cffbf79274.png)

It works great for the single line, but it may require making a method with a different name to avoid being a nuisance to the vararg overload callers :/

Reply:
Gif would be helpful to understand how this would be useful. What makes this throw off the varargs method? They're already different method names, but why should that even matter?